mvpa2/tests/test_generators.py

```diff
@@ -248,6 +248,18 @@ def test_balancer():
                  [4] * 2)
 
 
+def test_noncontiguous():
+    ds = give_data()
+    dist = 2
+    chunk = 4
+    nc = NonContiguous(dist=2)
+    ds.sa['partitions'] = (ds.chunks==chunk)+1
+    res = nc(ds)
+    trim_mask = np.logical_and(ds.sa.partitions==1, np.abs(ds.chunks-chunk)<=dist)
+    assert_array_equal( res.sa.partitions[trim_mask], 3)
+    assert_array_equal( res.sa.partitions[np.logical_not(trim_mask)],
+                        ds.sa.partitions[np.logical_not(trim_mask)])
+
```

Comment:
given the amount of options I feel that one testing scenario would be too optimistic ... could you come up with at least one which uses different from default options for which partitions to check etc? 

while at it I would appreciate if you could prettify the formatting a bit so it gets closer to recommended by PEP8. (e.g. no spaces after ```(``` in above.  I would recommend PyCharm IDE -- really neat one.  In NeuroDebian we have now pycharm-community-sloppy   package so you could just apt-get install it happen you use NeuroDebian.

Cheers and thanks for all your work!!!